### ISSUE TITLE
Null Pointer Exception when trying to open a connection

### ISSUE THREAD

**Issue by tiagosmx** [impossibl/pgjdbc-ng#327]
I'm getting a Null Pointer exception when trying to open a connection like this using pgjdbc-ng 0.7 version (No exception is thrown when I use pgjdbc-ng 0.6):
```
    public static void main(String[] args) {
        PGDataSource ds = new PGDataSource();
        ds.setHost("localhost");
        ds.setPort(5432);
        ds.setDatabase("postgres");
        ds.setUser("postgres");
        ds.setPassword("xxxxxxx");
        try {
            PGConnection con = (PGConnection) ds.getConnection();
        } catch (SQLException ex) {
            System.out.println("SQL Exception thrown: "+ ex);
        }
}
```
Error in 0.7 version:
![0 7](https://cloud.githubusercontent.com/assets/4307437/25839672/0811032a-346f-11e7-9b7b-9214f4441109.PNG)
Everything fine in 0.6 version:
![0 6](https://cloud.githubusercontent.com/assets/4307437/25839671/080fb006-346f-11e7-98f4-9ccd3d42d70c.PNG)
Tested in JDK 1.7

**Comment by jgeraerts** [comment]
Bumped into it as well. It was fixed in #307 but not released yet. You can clone the repository and create a build yourself.

**Comment by AhmedMourad0** [comment]
Any idea when it's being fixed? have same problem with version 0.7.1 here...

**Comment by tiagosmx** [comment]
@AhmedMouradDev so, I have managed to fix this by using maven on my project and importing the pgjdbc-ng library through maven.
I guess that those .jars provided on the website have dependency issues, that's why they throw null pointer exceptions.
My recommendation is using the lastest version provided in maven repository (that's what I'm using nowdays)
https://mvnrepository.com/artifact/com.impossibl.pgjdbc-ng/pgjdbc-ng

**Comment by AhmedMourad0** [comment]
@tiagosmx That's what i've been using, was using this Gradle dependency and got the error:

`compile group: 'com.impossibl.pgjdbc-ng', name: 'pgjdbc-ng', version: '0.7.1'`

However by using version 0.6 the error disappears but it's very annoying using an outdated version!

**Comment by sayseakleng** [comment]
@AhmedMouradDev , they haven't released the fix yet :(
We're waiting for the release :)

**Comment by tiagosmx** [comment]
@AhmedMouradDev @sayseakleng you sure you did not add the library manually instead of by maven/gradle?
I managed to get it working after adding pgjdbc-ng library through maven.

**Comment by kdubb** [comment]
Closing this as the issue appears to be fixed but lacked a specific release. It will for sure be included in the next release (0.8). Which should be out soon.
